pkg/hooks/monitor.go

```diff
@@ -48,18 +49,47 @@ func (m *Manager) Monitor(ctx context.Context, sync chan<- error) {
 	for {
 		select {
 		case event := <-watcher.Events:
-			if event.Op&fsnotify.Remove == fsnotify.Remove {
-				ok := m.remove(filepath.Base(event.Name))
-				if ok {
-					logrus.Debugf("removed hook %s", event.Name)
+			filename := filepath.Base(event.Name)
+			if len(m.directories) <= 1 {
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					ok := m.remove(filename)
+					if ok {
+						logrus.Debugf("removed hook %s", event.Name)
+					}
+				} else if event.Op&fsnotify.Create == fsnotify.Create || event.Op&fsnotify.Write == fsnotify.Write {
+					err = m.add(event.Name)
+					if err == nil {
+						logrus.Debugf("added hook %s", event.Name)
+					} else if err != ErrNoJSONSuffix {
+						logrus.Errorf("failed to add hook %s: %v", event.Name, err)
+					}
 				}
-			}
-			if event.Op&fsnotify.Create == fsnotify.Create || event.Op&fsnotify.Write == fsnotify.Write {
-				err = m.add(event.Name)
-				if err == nil {
-					logrus.Debugf("added hook %s", event.Name)
-				} else if err != ErrNoJSONSuffix {
-					logrus.Errorf("failed to add hook %s: %v", event.Name, err)
+			} else if event.Op&fsnotify.Create == fsnotify.Create || event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Remove == fsnotify.Remove {
+				err = nil
+				found := false
+				for i := len(m.directories) - 1; i >= 0; i-- {
+					path := filepath.Join(m.directories[i], filename)
+					err = m.add(path)
+					if err == nil {
+						found = true
+						logrus.Debugf("(re)added hook %s (triggered activity on %s)", path, event.Name)
+						break
+					} else if err == ErrNoJSONSuffix {
+						found = true
+						break // this is not going to change for fallback directories
+					} else if os.IsNotExist(err) {
+						continue // move on to the next fallback directory
+					} else {
+						found = true
+						logrus.Errorf("failed to (re)add hook %s (triggered by activity on %s): %v", path, event.Name, err)
+						break
+					}
+				}
+				if (found || event.Op&fsnotify.Remove == fsnotify.Remove) && err != nil {
+					ok := m.remove(filename)
```

Comment:
Nit: Single line

if ok := m.remove(filename); ok {
}

Reply:
> Nit: Single line

This is one instance of a more widespread pattern.  Lets sort it out [over here][1], and then I'll come back and apply the plan to this PR.

[1]: https://github.com/projectatomic/libpod/pull/758#discussion_r187769157

Reply:
> > Nit: Single line
>
> This is one instance of a more widespread pattern. Lets sort it out [over here][1], and then I'll come back and apply the plan to this PR.

Do we want to repeat the "[punt for now with possible follow-up work][2]" approach here as well?

[1]: https://github.com/projectatomic/libpod/pull/758#discussion_r187769157
[2]: https://github.com/projectatomic/libpod/pull/758#discussion_r188106308